app/jobs/import_versions_job.rb

```diff
@@ -174,4 +180,16 @@ def validate_kind!(kinds, record, field_name)
       raise Api::InputError, "`#{field_name}` must be #{names}, not `#{value.class.name}`"
     end
   end
+
+  def version_changed?(version)
+    return true if version.version_hash.nil?
+
+    previous = version.page.versions
+      .where(source_type: version.source_type)
+      .where('capture_time < ?', version.capture_time)
+      .order(capture_time: :desc)
```

Comment:
I noticed that, in `update_different_attribute`, you use `limit(1).first` which I assume is a performance win. Does that not buy you anything in this case?